research-hub-web/src/app/components/subhubs/subhubs.component.spec.ts

```diff
@@ -5,31 +5,439 @@ import { ApolloTestingController, ApolloTestingModule } from 'apollo-angular/tes
+              "slug": "super-dooper-research-service",
+              "title": "Super dooper research service",
+              "ssoProtected": true,
+              "summary": "The best service ever. It saved my life."
+            },
+            {
+              "__typename": "SubHub",
+              "slug": "a-subhub-to-be-included-as-level-2-of-a-landing-page-subhub",
+              "title": "a subhub to be included as level 2 of a landing page subhub",
+              "ssoProtected": true,
+              "summary": "as the title says"
+            },
+            {
+              "__typename": "Article",
+              "slug": "top-secret-article",
+              "title": "Top Secret Article",
+              "ssoProtected": true,
+              "summary": "For testing SSO"
+            },
+            {
+              "__typename": "Article",
+              "slug": "first-article",
+              "title": "First article",
+              "ssoProtected": false,
+              "summary": "A brief description of the first article. I'm writing some more stuff here just so that this seems a little more realistic. Sam was here. Have a good day."
+            }
+          ],
+          "__typename": "SubHubSubhubPagesCollection"
+        },
+        "__typename": "SubHub"
+      }
+    ],
+    "__typename": "SubHubCollection"
+  } as SubHubCollection);
 
   beforeEach(async(() => {
     TestBed.configureTestingModule({
       declarations: [SubhubsComponent],
       imports: [
         ApolloTestingModule,
+        CommonModule,
+        MaterialModule,
+        SharedModule,
+        BrowserAnimationsModule,
         RouterModule.forRoot([])
+      ], providers: [
+        AllSubHubChildPagesGQL,
+        AllContentItemParentSubHubsGQL
       ]
     })
       .compileComponents();
   }));
 
   beforeEach(() => {
+    controller = TestBed.get(ApolloTestingController);
+    subHubSpy = spyOn(SubhubsComponent.prototype, 'getSubHub').and.returnValue(currentSubHubData$);
+    allSubHubsSpy = spyOn(SubhubsComponent.prototype, 'getAllSubHubs').and.returnValue(currentSubHubData$);
+
     backend = TestBed.get(ApolloTestingController);
     fixture = TestBed.createComponent(SubhubsComponent);
     component = fixture.componentInstance;
     fixture.detectChanges();
   });
 
+  afterEach(() => {
+    controller.verify();
+  });
+
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  it('Title for a subhub page collection item should be truthy.', async () => {
+    const de = fixture.debugElement.query(By.css('#title'));
+    // expect(de.nativeElement.innerHTML).toEqual("Landing page for a 'sub-hub'");
+    // console.log(de);
+    expect(de.nativeElement.innerHTML).toBeTruthy();
+  });
+
+  xit('Should contain a parent subhub', () => {
+    let d = document.getElementsByClassName('parent-title');
+    console.log(d);
+
+    let e = fixture.debugElement.query(By.css('#parent-title'));
+    console.log(e);
+    expect(e).toBeTruthy();
```

Comment:
Remove console.log

Reply:
I'll remove that whole test since that code isn't used anymore.